hathor/p2p/sync_v2/manager.py

```diff
@@ -0,0 +1,1207 @@
+            self.send_data(tx)
+
+    def is_started(self) -> bool:
+        return self._started
+
+    def start(self) -> None:
+        """ Start sync.
+        """
+        if self._started:
+            raise Exception('NodeSyncBlock is already running')
+        self._started = True
+        self._lc_run.start(5)
+
+    def stop(self) -> None:
+        """ Stop sync.
+        """
+        if not self._started:
+            raise Exception('NodeSyncBlock is already stopped')
+        self._started = False
+        self._lc_run.stop()
+
+    def get_cmd_dict(self) -> dict[ProtocolMessages, Callable[[str], None]]:
+        """ Return a dict of messages of the plugin.
+
+        For further information about each message, see the RFC.
+        Link: https://github.com/HathorNetwork/rfcs/blob/master/text/0025-p2p-sync-v2.md#p2p-sync-protocol-messages
+        """
+        return {
+            ProtocolMessages.GET_NEXT_BLOCKS: self.handle_get_next_blocks,
+            ProtocolMessages.GET_PREV_BLOCKS: self.handle_get_prev_blocks,
+            ProtocolMessages.BLOCKS: self.handle_blocks,
+            ProtocolMessages.BLOCKS_END: self.handle_blocks_end,
+            ProtocolMessages.GET_BEST_BLOCK: self.handle_get_best_block,
+            ProtocolMessages.BEST_BLOCK: self.handle_best_block,
+            ProtocolMessages.GET_BLOCK_TXS: self.handle_get_block_txs,
+            ProtocolMessages.GET_TRANSACTIONS_BFS: self.handle_get_transactions_bfs,
+            ProtocolMessages.TRANSACTION: self.handle_transaction,
+            ProtocolMessages.TRANSACTIONS_END: self.handle_transactions_end,
+            ProtocolMessages.GET_PEER_BLOCK_HASHES: self.handle_get_peer_block_hashes,
+            ProtocolMessages.PEER_BLOCK_HASHES: self.handle_peer_block_hashes,
+            ProtocolMessages.STOP_BLOCK_STREAMING: self.handle_stop_block_streaming,
+            ProtocolMessages.GET_TIPS: self.handle_get_tips,
+            ProtocolMessages.TIPS: self.handle_tips,
+            ProtocolMessages.TIPS_END: self.handle_tips_end,
+            # XXX: overriding ReadyState.handle_error
+            ProtocolMessages.ERROR: self.handle_error,
+            ProtocolMessages.GET_DATA: self.handle_get_data,
+            ProtocolMessages.DATA: self.handle_data,
+            ProtocolMessages.RELAY: self.handle_relay,
+            ProtocolMessages.NOT_FOUND: self.handle_not_found,
+        }
+
+    def handle_not_found(self, payload: str) -> None:
+        """ Handle a received NOT-FOUND message.
+        """
+        # XXX: NOT_FOUND is a valid message, but we shouldn't ever receive it unless the other peer is running with a
+        #                modified code or if there is a bug
+        self.log.warn('not found? close connection', payload=payload)
+        self.protocol.send_error_and_close_connection('Unexpected NOT_FOUND')
+
+    def handle_error(self, payload: str) -> None:
+        """ Override protocols original handle_error so we can recover a sync in progress.
```

Comment:
We don't have this functionality to recover a sync in progress yet, right?

Should there be a TODO in this method to implement it?

Reply:
I'll remove this method and turn it into an issue/task.